src/expressions/operators/compares/nodeCompare.ts

```diff
@@ -1,66 +1,59 @@
 import DomFacade from '../../../domFacade/DomFacade';
+import DynamicContext from '../../../expressions/DynamicContext';
 import { compareNodePositions } from '../../dataTypes/documentOrderUtils';
 import ISequence from '../../dataTypes/ISequence';
 import isSubtypeOf from '../../dataTypes/isSubtypeOf';
-import sequenceFactory from '../../dataTypes/sequenceFactory';
 import { ValueType } from '../../dataTypes/Value';
-import zipSingleton from '../../util/zipSingleton';
 import arePointersEqual from './arePointersEqual';
 
 export default function nodeCompare(
 	operator: string,
 	domFacade: DomFacade,
-	firstSequence: ISequence,
-	secondSequence: ISequence
-): ISequence {
+	first: ValueType,
+	second: ValueType
+): (a: ISequence, b: ISequence, context: DynamicContext) => boolean {
```

Comment:
Some docs would be awesome here? a? b? Some more descriptive names would be great.

Reply:
Named switched to something more descriptive and added documentations on top of the function.